File: c/to-camel-case.c
/**
 * Tanggal: 23 Februari 2026
 * Link: https://www.codewars.com/kata/517abf86da9663f1d2000003/train/c
 */

#include <ctype.h>
#include <stdbool.h>
#include <stddef.h>
#include <stdio.h>
#include <string.h>

void to_camel_case(const char *text, char *camel) {
    size_t length = strlen(text);

    if (length == 0) {
        camel[0] = '\0';
        return;
    }

    int i = 0;
    int j = 0;
    bool shouldBeCapitalized = false;

    while (i < length) {
        if (text[i] == '-' || text[i] == '_') {
            i++;
            shouldBeCapitalized = true;
            continue;
        }

        if (shouldBeCapitalized) {
            camel[j] = (char)toupper(text[i]);
            shouldBeCapitalized = false;
        } else {
            camel[j] = text[i];
        }

        i++;
        j++;
    }

    camel[j] = '\0';
}

int main() {
    const char text[] = "above-large-Mother";
    size_t length = strlen(text);
    char camel[length + 1];

    to_camel_case(text, camel);
    printf("%s\n", camel);

    return 0;
}

File: java/ToCamelCase.java
/**
 * Tanggal: 11 Agustus 2026
 * Link: https://www.codewars.com/kata/517abf86da9663f1d2000003/train/java
 */

public class ToCamelCase {
    public static void main(String[] args) {
        System.out.println(toCamelCase("The_Stealth-Warrior"));
    }

    public static String toCamelCase(String s) {
        String result = "";
        boolean shouldBeCamelCased = false;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '-' || s.charAt(i) == '_') {
                shouldBeCamelCased = true;
                result += " ";
                continue;
            }

            if (shouldBeCamelCased) {
                shouldBeCamelCased = false;
                result += String.valueOf(s.charAt(i)).toUpperCase();
                continue;
            }

            result += String.valueOf(s.charAt(i));
        }

        return result.replaceAll(" ", "");
    }
}
